src/cephadm/cephadm

```diff
@@ -3204,10 +3231,58 @@ def list_daemons(detail=True, legacy_dir=None):
                         i['configured'] = get_file_timestamp(
                             os.path.join(data_dir, fsid, j, 'unit.configured'))
 
+                    #add the dict object for this container to list
                     ls.append(i)
 
     return ls
 
+# turns go template output into 2d array
+# out parameter is go template string output from command
+# n paramater is number of arguments in format of command
+# Ex: --format='{{.ID}},{{.Image}},{{.Names}}' would require n = 3
+def parse_go_output(out, n):
+    # command outputs as a single string. Converting it into a 2d
+    # array of strings by splitting on newlines and commas
+    out_lines = out.strip().split('\n')
+    # initialize 2d array to have # of rows equal to rows in output and
+    # n columns: one for each data item it provides.
+    parsed_out = [["a" for placeholder in range(n)] for placeholder in range(len(out_lines))]
+    for i in range(0, len(out_lines)):
+        single_line = out_lines[i].split(',')
+        for j in range(0, n):
+            parsed_out[i][j] = single_line[j].replace("'", "")
+    return parsed_out
+
+# helper function for list_daemons that takes parsed ps output and parsed image ls output
+# and returns container id, image id, image name and start timestamp.
+# Assumes parsed output is 2d array formatted a specific way
+def get_list_daemons_data(unit_name, parsed_ps_out, parsed_image_list_out):
+    container_id = None
+    image_name = None
+    image_id = None
+    start_stamp = None
+    for k in range(0, len(parsed_ps_out)):
+        # match container name from get_unit_name with name in docker ps command to
+        # find that container's info
+        if unit_name in parsed_ps_out[k][2]:
+            # grab container id, image name and start timestamp from corresponding row in ps output
+            container_id = parsed_ps_out[k][0]
+            image_name = parsed_ps_out[k][1]
+            start_stamp = parsed_ps_out[k][3]
+    # if an image name was found, can use it to find that image's id
+    if image_name:
+        # image names from docker image list command do not include extra
+        # info sometimes found after a colon. Ex: "latest-master-devel" in
+        # "docker.io/ceph/daemon-base:latest-master-devel". Adjusted version removes any
+        # possible info after a colon so strings match
+        adjusted_image_name = image_name.split(':')[0]
+        # match found image name with image name in podman image list output
+        for k in range(0, len(parsed_image_list_out)):
+            if adjusted_image_name in parsed_image_list_out[k][0]:
+                image_id = parsed_image_list_out[k][1]
```

Comment:
This also needs to filter the list by both image repository and tag...


In the below, `container_image_id` is incorrectly listed as mimic ..

```
$ cephadm ls
    {
        "style": "cephadm:v1",
        "name": "crash.host1",
        "fsid": "3dffcc91-9a67-4703-855f-eb1593e624f8",
        "systemd_unit": "ceph-3dffcc91-9a67-4703-855f-eb1593e624f8@crash.host1",
        "enabled": true,
        "state": "running",
        "container_id": "5b184588ff45",
        "container_image_name": "docker.io/ceph/daemon-base:latest-master-devel",
        "container_image_id": "b14439ab0438",
        "version": "16.0.0-899-g18734af",
        "started": "2020-06-08 16:33:56.253103622 -0600 MDT",
        "created": "2020-06-08T22:33:56.003487",
        "deployed": "2020-06-08T22:33:54.027485",
        "configured": "2020-06-08T22:33:56.003487"
    },

$ podman image ls | grep daemon-base
docker.io/ceph/daemon-base                                                latest-master-devel                        7bb50d3c5c9a   7 weeks ago     1.12 GB
docker.io/ceph/daemon-base                                                latest-octopus                             dc4b346b2f95   2 months ago    1.12 GB
docker.io/ceph/daemon-base                                                latest-nautilus                            d45269756fb0   3 months ago    860 MB
docker.io/ceph/daemon-base                                                latest-mimic                               b14439ab0438   3 months ago    607 MB
```